examples/demo-stf/README.md

```diff
@@ -0,0 +1,149 @@
+
+This package shows how you can combine modules to build a custom state transition function. We provide several module implementations
+for you, and if you want additional functionality you can find a tutorial on writing custom moduleshere (TODO: Insert link!).
+
+For purposes of this tutorial, the exact choices of modules don't matter at all - the steps to combine modules are identical
+no matter which ones you pick.
+
+## Overview
+
+To get a fully functional rollup, we recommend implementing two traits. First, there's the [State Transition Function
+interface](../../rollup-interface/specs/interfaces/stf.md) ("STF") , which specifies your rollup's abstract logic. Second, there's
+a related trait called `State Transition Runner` ("STR") which tells a full node how to instantiate your abstract STF on a concrete machine.
+
+Strictly speaking, it's sufficient for a rollup to only implement the first interface. If you've done that, it's possible to integrate
+with ZKVMs and DA Layers - but you'll have to customize your full node implementation a bit to deal with your particular rollup's
+configuration. By implementing the STR trait, we make it much easier for the full-node implementation to understand how to interact
+with the rollup generically - so we can keep our modifications as minimal as possible. In this demo, we'll implement both traits.
+
+## Implementing State Transition _Function_
+
+As you recall, the module system is primarily designed to help you implement the [State Transition Function
+interface](../../rollup-interface/specs/interfaces/stf.md).
+
+That interface is quite high-level - the only notion
+that it surfaces it that of a `blob` of rollup data. In the module system, we work at a much lower level - with
+transactions signed by particular private keys. To bridge the gap, there's a system called an `AppTemplate`, which
+translates between the two layers of abstraction.
+
+The reason the `AppTemplate` is called a "template" is that it's generic. It allows you, the developer, to pass in
+several parameters which specify its exact behavior. In order, these four generics are
+
+1. `Context`: a per-transaction struct containing the message's sender. This also provides handles to cryptographic operations
+2. `TxVerifier`: verifies the signatures on transaction and deserializes them into messages
+3. `Runtime`: a collection of modules which make up the rollup's public interface
+4. `TxHooks`: a set of functions which are invoked at various points in the transaction lifecycle
+
+To implement your state transition function, you simply need to specify values for each of these four fields.
+
+So, a typical app definition looks like this:
+
+```rust
+pub type MyNativeStf = AppTemplate<DefaultContext, MyTxVerifier<DefaultContext>, MyRuntime<DefaultContext>, MyTxHooks<DefaultContext>>;
+pub type MyZkStf = AppTemplate<ZkDefaultContext, MyTxVerifier<DefaultContext>, MyRuntime<DefaultContext>, MyTxHooks<DefaultContext>>;
```

Comment:
It should be `ZkDefaultContext ` also for the other structures `MyTxVerifier` etc..

Reply:
Fixed!